libcodechecker/server/api/report_server.py

```diff
@@ -396,31 +409,54 @@ def is_touched(self, run_id, report_id):
             return report_id in self.__sessions[run_id]['touched_reports']
 
         def has_ongoing_run(self, run_id):
+            LOG.error(self.__sessions)
             return run_id in self.__sessions
 
         def get_transaction(self, run_id):
-            self.__sessions[run_id]['timer'] = time.time()
-            return self.__sessions[run_id]['transaction']
-
-        def _timeout_sessions(self):
+            """
+            Retrieve the SQLAlchemy session that encapsulated the transaction
+            for the given run_id's store operation.
+            """
+            LOG.error("get_transaction(" + str(run_id) + ")")
+            session = self.__sessions[run_id]
+            session['timer'] = time.time()
+
+            # Because the run was "accessed" by a store operation, we update
+            # the run's timestamp.
+            if int(time.time() - session['last_lock_update']) > 5 * 60:
```

Comment:
What are these magic constants?

Reply:
5 minutes.

Reply:
@Xazax-hun There will be an undertaking which aims to move all limit-like variables such as store length, data size out of the code and into a configuration file which is much better fine-tuneable per individual server deployments based on the servers' capabilities and the needs of the users of that server.